spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -424,6 +423,7 @@ public final void stop(boolean wait) {
 				else {
 					doStop(() -> { });
 				}
+				publishContainerStoppedEvent();
```

Comment:
Is it really stopped at this point?

Don't we really emit such an event from that `callback` we provide for the `doStop()` ?